hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -406,6 +407,9 @@ public void onResponse(Data response) {
                 public void onFailure(Throwable t) {
                 }
             });
+            if (getMapConfig().isStatisticsEnabled()) {
```

Comment:
Sure. But I always get a bit frustrated when hard work gets undone. 

Actually the problem is bigger than 1 object; actually it is 2. Also the node object in the invocation future is created. I have this on my list for 3.7; so optimization for invocation future with just a single callback.